### DISCUSSION TITLE
Allow creation of all tables as unlogged in non-production environments

### DISCUSSION THREAD

**Discussion by JacobEvelyn** [jeremyevans/sequel#2134] [General]
Postgres provides guidance on [settings](https://www.postgresql.org/docs/current/non-durability.html) that can be tweaked to reduce durability when not needed. One of these is the use of unlogged tables. We'd like to be able to speed up our DB-heavy tests by having our Rails app use only unlogged tables in test environments, which we set up via `rake db:schema:load` loading a `schema.rb` file.

I'm aware that in #1778 you indicated the schema file should not include `unlogged: true`. However, would you be open to a global, opt-in option we could configure in e.g. `config/environments/test.rb` allowing us to force all tables to be created as unlogged?

For comparison, `ActiveRecord` [supports this](https://github.com/rails/rails/pull/34221) via the `ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.create_unlogged_tables` setting.

**Comment by jeremyevans** [comment]
I'll consider a Database option on PostgreSQL to create unlogged tables by default (if no `:unlogged` option is given).

**Reply by JacobEvelyn** [reply]
Thanks @jeremyevans ! I'm not too familiar with the `sequel` codebase. Could you give me some guidance on where to start implementing that change?

**Reply by jeremyevans** [reply]
`git grep :unlogged` will show where the `:unlogged` option is currently used.  You can use `@opts[:unlogged_tables_default]` to access a Database option there.

**Comment by jeremyevans** [comment]
I added support for this in 235a62a53d954f7fb1e133655d6732215040c96c

**Reply by JacobEvelyn** [reply]
Thank you!
